packages/telestion-client-common/src/hooks/stores/use-dialog.model.ts

```diff
@@ -0,0 +1,91 @@
+import { ReactNode } from 'react';
+import { State } from 'zustand';
+
+/**
+ * A configuration object to customize a dialog
+ * created via the {@link DialogState.show} function.
+ */
+export interface DialogConfig {
+	/**
+	 * The title of the dialog displayed in the upper left corner.
+	 * @see {@link https://react-spectrum.adobe.com/react-spectrum/Dialog.html#content}
+	 */
+	title: ReactNode;
+
+	/**
+	 * An optional header of the dialog displayed in the upper right corner.
+	 * @see {@link https://react-spectrum.adobe.com/react-spectrum/Dialog.html#content}
+	 */
+	header?: string | ReactNode;
+
+	/**
+	 * The content of the dialog rendered below the title and header
+	 * in the center of the dialog.
+	 * @see {@link https://react-spectrum.adobe.com/react-spectrum/Dialog.html#content}
+	 */
+	content: string | ReactNode;
+
+	/**
+	 * An optional footer of the dialog in the lower left corner.
+	 * @see {@link https://react-spectrum.adobe.com/react-spectrum/Dialog.html#content}
+	 */
+	footer?: ReactNode;
+}
+
+/**
+ * A dialog which is registered in the Telestion Common dialog state.
+ *
+ * Every dialog has an id which is unique inside the project.
+ */
+export interface Dialog {
+	/**
+	 * The unique id of the dialog
+	 * given via the {@link DialogState.show} function.
+	 */
+	id: string;
+
+	/**
+	 * The configuration of the dialog
+	 * given via the {@link DialogState.show} function.
+	 */
+	config: DialogConfig;
+
+	/**
+	 * When `true`, the dialog is currently open.
+	 */
+	isOpen: boolean;
+
+	/**
+	 * A handler to confirm the opened dialog.
+	 */
+	onConfirm: () => void;
+
+	/**
+	 * A handler to cancel the opened dialog.
+	 */
+	onCancel: () => void;
+}
+
+/**
+ * The dialog store and actions of Telestion Client Common.
+ *
+ * Stores and handles show dialog calls from projects and common itself.
+ *
+ * @see {@link DialogState.show}
+ */
+export interface DialogState extends State {
+	/**
+	 * Stores the currently registered dialogs.
+	 * @see {@link Modal}
+	 */
+	dialogs: Array<Dialog>;
+
+	/**
+	 * Registers and opens a new dialog configured with the specified options.
+	 * @param id - the id of the modal, should be unique inside the project
+	 * @param config - the specified options to customize the dialog
+	 * @returns a Promise which resolves, when the user confirmed
+	 * and rejects, when the user canceled the dialog
+	 */
+	show(id: string, config: DialogConfig): Promise<void>;
```

Comment:
I'm thinking about whether there's a simple way to make this a `Promise<DialogResult>` instead of a `Promise<void>`.

As it is, we're really only supporting a `confirm()`/`alert()`-like use case. I think without too much more effort, it should be possible to support generic dialogs that then reject on cancel and resolve with the dialog's arbitrary result (e.g., form input inside the dialog)...

Reply:
For example, giving a state to the `content` element in the configuration? :thinking: 

```tsx
const config: DialogConfig<string> = {
	title: 'My Advanced Dialog',
	content: (state, setState) => (
		<Flex>
			<Label>What's your name?</Label>
			<TextField value={state} onChange={setState} />
		</Flex>
	)
};

// setup and call
```

Reply:
Where `<string>` is a generic type for the `state` in the content function and the return type of the promise.

Reply:
Exactly 😉

Reply:
Done in [`12d5618` (#855)](https://github.com/wuespace/telestion-client/pull/855/commits/12d5618bec942114af8cecaf918521ddf4834ae6)